### PR TITLE
Add TypeScript Definition

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+declare class Hasher {
+    constructor(size?: 224 | 256 | 384 | 512);
+
+    update(data: Buffer | string): this;
```
Comment (septs) [inline index.d.ts]: The `update` function, not supported `encoding` arguments

https://github.com/phusion/node-sha3/blob/master/src/index.js#L11-L22
The `update` function, This modification can to be consistent with the documentation
```diff
diff --git a/src/index.js b/src/index.js
index 477ed82..8f1eb57 100644
--- a/src/index.js
+++ b/src/index.js
@@ -8,14 +8,14 @@ const createHash = ({ padding }) => function Hash(size = 512) {
 
   const sponge = new Sponge({ capacity: size, padding });
 
-  this.update = (input) => {
+  this.update = (input, encoding = 'utf8') => {
     if (Buffer.isBuffer(input)) {
       sponge.absorb(input);
       return this;
     }
 
     if (typeof input === 'string') {
-      return this.update(Buffer.from(input, 'utf8'));
+      return this.update(Buffer.from(input, encoding));
     }
 
     throw new TypeError('Not a string or buffer');
```

see https://github.com/phusion/node-sha3#updatedata-encodingutf8

**Comment by septs** [comment]
About implementing inconsistencies with documentation.

Documentation: https://github.com/phusion/node-sha3#updatedata-encodingutf8
Implemention: https://github.com/phusion/node-sha3/blob/master/src/index.js#L11-L22

The `update` function, This modification can to be consistent with the documentation
```diff
diff --git a/src/index.js b/src/index.js
index 477ed82..8f1eb57 100644
--- a/src/index.js
+++ b/src/index.js
@@ -8,14 +8,14 @@ const createHash = ({ padding }) => function Hash(size = 512) {
 
   const sponge = new Sponge({ capacity: size, padding });
 
-  this.update = (input) => {
+  this.update = (input, encoding = 'utf8') => {
     if (Buffer.isBuffer(input)) {
       sponge.absorb(input);
       return this;
     }
 
     if (typeof input === 'string') {
-      return this.update(Buffer.from(input, 'utf8'));
+      return this.update(Buffer.from(input, encoding));
     }
 
     throw new TypeError('Not a string or buffer');
```

File: package.json [hunk 2]
```diff
@@ -34,11 +35,12 @@
     "@babel/core": "7.1.5",
     "@babel/preset-env": "7.1.5",
     "@babel/register": "7.0.0",
+    "@types/node": "^11.11.3",
     "babel-eslint": "10.0.1",
     "eslint": "5.9.0",
     "eslint-plugin-ante": "1.0.1",
     "mocha": "5.2.0",
-    "nyc": "13.1.0"
+    "nyc": "^13.3.0"
```
Comment (canterberry) [inline package.json]: :lock: Prefer locked versions over fuzzy versions. Instead of `^13.3.0`, should be `13.3.0`.

File: package.json [hunk 3]
```diff
@@ -34,11 +35,12 @@
     "@babel/core": "7.1.5",
     "@babel/preset-env": "7.1.5",
     "@babel/register": "7.0.0",
+    "@types/node": "^11.11.3",
```
Comment (canterberry) [inline package.json]: :lock: Prefer locked versions over fuzzy versions. Instead of `^11.11.3`, should be `11.11.3`.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/node_modules/
```
Comment (canterberry) [inline .gitignore]: :memo: Please add a comment above this line (like in the entries below), something to the effect of:

```
# Generated by `yarn install`
```

**Comment by septs** [comment]
@canterberry request re-review

**Comment by canterberry** [comment]
Changes look good, @septs, but one more thing I missed earlier. This repo requires [signed commits](https://help.github.com/en/articles/signing-commits). My apologies for the back and forth -- I should have noticed in the first review.

Would you mind signing your commits on this PR? Alternatively, and with your permission, I can replicate these changes and push up signed commits, myself (with attribution to you).

**Comment by septs** [comment]
OK

**Comment by septs** [comment]
@canterberry README file may need to update accordingly.

**Comment by septs** [comment]
@canterberry request merge to master

**Comment by canterberry** [comment]
Alrighty, I've pushed up #49. That PR contains an identical changeset to this PR, but with signed commits so we can get it merged in. I rebased and squashed the commits, retaining attribution to @septs on the original work. Closing this PR in favor of that one.

**Pull request closed without merging**